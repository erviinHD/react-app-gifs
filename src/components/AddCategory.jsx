import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return;
    // setCategories((categories) => [...categories, inputValue]);
    onNewCategory(inputValue.trim());
    setInputValue("");
  };

  const onClean = () => {
    window.location.reload();
  };

  return (
    <>
      <div className="display-form">
        <form onSubmit={onSubmit} aria-label="form">
          <input
            type="text"
            placeholder="Search Gifs"
            value={inputValue}
            onChange={onInputChange}
          />
        </form>
        <button onClick={onClean} className="btn-reload">
          x
        </button>
      </div>
    </>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
