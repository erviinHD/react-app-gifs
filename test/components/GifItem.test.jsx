import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";

describe("Test to GifItem component", () => {
  const title = "Saitama";
  const url = "http://localhost/Saitama";
  test("should to be match with snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("should be show img url and alt", () => {
    render(<GifItem title={title} url={url} />);
    //screen.debug()
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(alt);
  });

  test("should be show a title in component", () => {
    render(<GifItem title={title} url={url} />);

    expect(screen.getByText(title)).toBeTruthy();
  });
});
