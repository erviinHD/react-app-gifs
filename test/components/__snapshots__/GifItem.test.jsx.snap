// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to GifItem component should to be match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="http://localhost/Saitama"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
